distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
```
// Treat octet stream as media type config
// see https://github.com/docker/distribution/issues/1621
if m.Manifest.Config.MediaType == "application/octet-stream" {
	m.Manifest.Config.MediaType = schema2.MediaTypeConfig
}
if m.Manifest.Config.MediaType != schema2.MediaTypeConfig {
	return false, errors.New("unsupported configuration type")
}
```

The first part could be anywhere but this would accomplish what I am talking about